### ISSUE TITLE
Support ActiveNode modules which can be included into classes

### ISSUE THREAD

**Issue by cheerfulstoic** [neo4jrb/neo4j#543]

**Comment by cheerfulstoic** [comment]
A usecase in discussed here (see comments): http://stackoverflow.com/questions/28838112/multiple-types-for-each-record-in-neo4j-rb/28854860?noredirect=1#comment46030873_28854860

**Comment by mmmmmrob** [comment]
I’m wanting to create a relationship that’s polymorphic in both directions and, ideally, accessible through a ruby accessor. Something like:
  (:Magazine)-[:mentions]->(:Celebrity)
  (:Celebrity)-[:mentions]->(:HolidayDestination)
and be able to access it like 'Celebrity.find(1234).mentions’

Having the inverse accessor HolidayDestination.find(7).mentioned_by would also be useful

I plan to do this with Modules and include as necessary. I would also like the module to label the classes as :Mentions and :Mentionable or something like that. An inheritance hierarchy doesn't allow for this case. Maybe something ActiveNode#self.AddLabel would allow the module to add a label from Module#self.included(klass)?

**Comment by subvertallchris** [comment]
Hey @mmmmmrob, that's interesting. Sorry for the slow response here, I've been on vacation and Brian's been moving.

I think that all we're missing to allow this is, like you said, a way to additional labels to existing classes. At the moment, `self.mapped_label_name=` will replace, which is no good. I have some thoughts about how this can be implemented that I'm going to play with now, we'll see if it gets us there.

**Comment by chaodoze** [comment]
+1 for allowing multiple non hierarchically related labels.
Another use case I have is to assign tags to different nodes (these could be colors, or hash tags or topic string tags) and they can apply to many different types of nodes

**Comment by cheerfulstoic** [comment]
Sorry for the delay, just getting back from vacation.

You can definitely have modules which provided shared functionality (including properties/associations/validations/etc...), but you're saying you'd want a label to go along with that?  Could you give an example?

**Comment by chaodoze** [comment]
Hi Brian, thanks for the response, I may have misunderstood this thread. I'm simply want to add more Labels to an exist node (that already has a Label). I think I can just do that with <node>.add_label(), right?

**Comment by cheerfulstoic** [comment]
Yeah, you can use `add_label` to add a new label.  Just keep in mind that that may affect how that node is loaded in the future (which is maybe what you want).  For example if you have a `Person` model and an `Author` model which inherits from `Person`, then adding an `Author` label to a `Person` node will cause it to be loaded as an `Author` node in the future (though, the node won't change class when you do that.  Perhaps that's something that we should look into).

Probably that's what you want, but it's good to keep in mind.  You might look at this spec to see how different labels combinations for a node are loaded as different models:

https://github.com/neo4jrb/neo4j/blob/master/spec/e2e/node_wrapping_spec.rb

This issue is about the idea that you might have, for example, a `Person` model which includes a `Commentable` module (which has been set up in a certain way) and that the module would also mean that `Person` nodes also get a `Commentable` label.  That example is a bit odd, and the currently open question is if anybody has a good example of when including modules and having them affect labels ever makes sense.
